engine/termops.ml

```diff
@@ -980,9 +980,9 @@ let rec strip_outer_cast sigma c = match EConstr.kind sigma c with
 (* flattens application lists throwing casts in-between *)
 let collapse_appl sigma c = match EConstr.kind sigma c with
   | App (f,cl) ->
-    if EConstr.isCast sigma f then
+    if EConstr.isApp sigma f then
```

Comment:
Isn't this impossible?

Reply:
What if f was an evar assigned to an App?